components/browser/engine-gecko-nightly/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngineSession.kt

```diff
@@ -373,6 +373,59 @@ class GeckoEngineSession(
         override fun onLoadRequest(
             session: GeckoSession,
             request: NavigationDelegate.LoadRequest
+        ): GeckoResult<AllowOrDeny> {
+            return handleLoadRequest(request)
+        }
+
+        override fun onSubframeLoadRequest(
+            session: GeckoSession,
+            request: NavigationDelegate.LoadRequest
+        ): GeckoResult<AllowOrDeny> {
+            return handleLoadRequest(request)
```

Comment:
One concern I have here is that the new `onSubframeLoadRequest` will trigger a `onLoadRequest` on all our engine observers. They will update the search terms, expand the toolbar etc. which all don't seem necessary if a sub frame is loaded. Should we introduce a separate observer method here?

I am also curious if these load requests are additional now or if the total number is the same, just split up between the two methods in GV? This is important as we tweaked performance of `onLoadRequest` and don't want to regress. If we add a separate engine observer method it would address that concern as well.

Reply:
After confirming with GV, you're correct that it will be more load requests in total.  I'll create a new engine observer to handle `onSubframeLoadRequest` API.  Thanks for catching this.